packages/blade/src/components/Dropdown/DropdownOverlay.web.tsx

```diff
@@ -60,13 +63,39 @@ type DropdownOverlayProps = {
  * Wrap your ActionList within this component
  */
 const _DropdownOverlay = ({ children, testID }: DropdownOverlayProps): JSX.Element => {
-  const { isOpen, triggererRef, hasLabelOnLeft, dropdownTriggerer } = useDropdown();
+  const { isOpen, triggererRef, hasLabelOnLeft, dropdownTriggerer, setIsOpen } = useDropdown();
   const { theme } = useTheme();
   const [display, setDisplay] = React.useState<'none' | 'block'>('none');
   const [width, setWidth] = React.useState<SpacingValueType>('100%');
+  const [dropdownPosition, setDropdownPosition] = useState<DropdownPosition>({});
 
   const isMenu = dropdownTriggerer !== 'SelectInput';
 
+  const detectOverflowMiddleware = {
```

Comment:
Possible to put this middle ware outside of the function so it doesn't gets recreated everytime? 

Maybe for the `setDropdownPosition` you can pass it via arguments